contentcuration/contentcuration/tests/test_authentication.py

```diff
@@ -0,0 +1,133 @@
+import datetime
+from contentcuration.utils.policies import check_policies
+
+from base import BaseTestCase
+
+
+class AuthenticationTestCase(BaseTestCase):
+    def setUp(self):
+        super(AuthenticationTestCase, self).setUp()
+        self.base_url = '/channels/{}'.format(self.channel.pk)
+        self.view_url = '{}/view'.format(self.base_url)
+        self.edit_url = '{}/edit'.format(self.base_url)
+
+    def test_authenticate_policy_update(self):
+        """
+        Test that authenticated new users are shown the policies page regardless of what page was requested
+        if they have policies they have not yet agreed to.
+        """
+        base_url = '/channels/{}'.format(self.channel.pk)
+
+        self.channel.viewers.add(self.user)
+        self.client.force_login(self.user)
+
+        assert len(check_policies(self.user)) > 0
+        # ensure that a new user is redirected to policy update after authenticating the first time.
+        response = self.client.get(base_url, follow=True)
+        assert "/policies/update" == response.redirect_chain[-1][0]
+
+    def test_staging_channel_access(self):
+        staging_url = '{}/staging'.format(self.base_url)
+
+        # test that when we are redirected we get taken to the login page since we're not signed in,
+        # and that after sign in we'll get sent to the right place.
+        response = self.get(staging_url, follow=True)
+        assert "/accounts/login/?next={}".format(staging_url) == response.redirect_chain[-1][0]
+        assert response.status_code == 200
+
+        self.sign_in()
+
+        response = self.get(staging_url)
+        assert response.status_code == 403
+
+        self.channel.viewers.add(self.user)
+
+        response = self.get(staging_url)
+        assert response.status_code == 403
+
+        self.channel.editors.add(self.user)
+
+        # finally!
+        response = self.get(staging_url)
+        assert response.status_code == 200
```

Comment:
I like your coding style with this test, it's very clean, and feels like reading a story!